### ISSUE TITLE
NRedisStack not writing timeseries values to Valkey

### ISSUE THREAD

**Issue by adimishraa7** [redis/NRedisStack#318]
Thanks for reporting an issue in NRedisStack! Please update the appropriate text below, as much data as possible really helps!

**NRedisStack Version**: 0.12.0

**Redis Stack Version**:  Valkey 7.2.5

**Description**: Unable to write timeseries values to valkey.
var timestamp = new TimeStamp(long.Parse(data_float[0]));
 var data_vals = new List<(string key, NRedisStack.DataTypes.TimeStamp, double value)>();
 data_vals.Add(("mag", timestamp, Convert.ToDouble(data_float[4])));
 data_vals.Add(("ang", timestamp, Convert.ToDouble(data_float[5])));
  IReadOnlyCollection<(string,NRedisStack.DataTypes.TimeStamp,double)>data_collection = data_vals.AsReadOnly(); ts.MAdd(data_collection);

**Comment by uglide** [comment]
Hello @adimishraa7 

NRedisStack is a Redis official client library. As a RESP-compatible library, it can work with forks without guaranteeing future compatibility due to API changes or deprecations. In particular, NRedisStack supports hash-field expiration and other Redis-only advanced features like indexed searches, the JSON data type, time series, and probabilistic data structures that other Redis forks or imitators do not offer.

Simply put, you need to use Redis to benefit from all the advanced features available in NRedisStack.
